mslice/cli/helperfunctions.py

```diff
@@ -154,6 +154,8 @@ def is_slice(*args):
     Checks if args[0] is a WorkspaceBase or HistogramWorkspace
     """
     if len(args) > 0:
+        if isinstance(args[0], HistogramWorkspace) and args[0].is_PSD:
```

Comment:
I'm not sure `is_PSD` is the correct check to use here. `is_PSD` just indicates whether the instrument is a PSD-tube-based instrument or not and is not really anything to do with the data type. The projection should return a `PixelWorkspace` I think so a slice should be able to be created if it is a `PixelWorkspace` or a `Workspace`

Reply:
I checked and it was a histogram workspace that is being passed as the argument

Reply:
That's odd as a projection should still be 2D and histogram is 1D.

Reply:
Actually I think `is_PSD` is correct here - we are dealing with the slice and not the project workspace itself. The slice is either an `MDHistoWorkspace` generated from `BinMD` if the parent is from a PSD instrument (parent is a `PixelWorkspace`) or a `MatrixWorkspace` generated from `SofQW3` if the parent is from a non-PSD instrument (parent is also a `MatrixWorkspace`).

Reply:
My bad for not remember that!
Should we really check the dimensionality of the data though rather than rely on the fact that for some more instrument-specific check?

Reply:
Yeah... a `getNumDims() == 2` check is probably better! It would still be valid in future if we should change the code somehow.